core/src/HIP/Kokkos_HIP_Instance.hpp

```diff
@@ -117,8 +117,8 @@ class HIPInternal {
   mutable std::mutex m_mutexSharedMemory;
 
   // Scratch Spaces for Reductions
-  size_type m_scratchSpaceCount = 0;
-  size_type m_scratchFlagsCount = 0;
+  std::size_t m_scratchSpaceCount = 0;
+  std::size_t m_scratchFlagsCount = 0;
 
   size_type *m_scratchSpace           = nullptr;
   size_type *m_scratchFlags           = nullptr;
```

Comment:
We basically say that the allocated size is a multiple of the size of `ScratchGrain` which is based on `ExecutionSpace::size`. IMHO, it would not make sense to use `std::size_t*` here (but maybe `void*` or `ScratchGrain`).
I think the discussion for what these are is somewhat separate from the type used for requesting scratch space.